src/mtype.d

```diff
@@ -5983,7 +5985,17 @@ public:
                         for (size_t j = 0; j < tdim; j++)
                         {
                             Parameter narg = (*tt.arguments)[j];
-                            (*newparams)[j] = new Parameter(narg.storageClass | fparam.storageClass, narg.type, narg.ident, narg.defaultArg);
+
+                            // Bugzilla 12744: If the storage classes of narg
+                            // conflict with the ones in fparam, it's ignored.
```

Comment:
I'm not so sure that ignoring the conflict is the right way to go. Seems to me that an error ought to be issued. If it later turns out that there is a compelling reason to allow it, we can always change it. But if we ignore the conflict, it is much harder to go back and make it an error.

I recommend making it an error.